### PR TITLE
Pipelining (rfc2920)

### PR DESCRIPTION
**Pull request by LangArthur**
## Requirements

- [X] I have read the `CODE_OF_CONDUCT.md`.
- [x] I have updated `CHANGELOG.md` & `tools/install/deb/changelog` to reflect my changes.

## Motivation

The pipelining support was missing.

### Added
- support for pipelining

### CODE REVIEW

**Review by ltabis — APPROVED** [review]

**Comment by LangArthur** [comment]
ready to go !

File: src/vsmtp/vsmtp-protocol/src/reader.rs [hunk 1]
```diff
@@ -121,40 +219,21 @@ impl<R: tokio::io::AsyncRead + Unpin + Send> Reader<R> {
 
     /// Produce a stream of ESMTP commands.
     #[inline]
-    #[allow(clippy::expect_used)]
     pub fn as_command_stream(
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-protocol/src/reader.rs:222]: Could be implemented as `self.as_line_stream().map_ok(parse_command_line)`

https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html

File: src/vsmtp/vsmtp-protocol/src/receiver.rs [hunk 2]
```diff
@@ -416,83 +416,88 @@ where
...
+                        }
+                        Error::ParsingError(e) => {
+                            return Err(std::io::Error::new(std::io::ErrorKind::InvalidData, e))
+                        }
+                    },
+                };
+                tracing::trace!("<< {:?} ; {:?}", verb, std::str::from_utf8(&args.0));
+
+                let stage = self.handler.get_stage();
+                let reply = match (&verb, &stage) {
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-protocol/src/receiver.rs:455]: Should be `(verb, stage)`, those enum are supposed to be Copyable

https://doc.rust-lang.org/std/marker/trait.Copy.html

File: src/vsmtp/vsmtp-server/src/receiver/pre_transaction.rs [hunk 3]
```diff
@@ -324,6 +324,7 @@ where
                         }),
                         Some("250-8BITMIME\r\n".to_string()),
                         Some("250 SMTPUTF8\r\n".to_string()),
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-server/src/receiver/pre_transaction.rs:326]: Must have `250-SMTPUTF8`

File: src/vsmtp/vsmtp-server/src/receiver/pre_transaction.rs [hunk 4]
```diff
@@ -365,6 +366,7 @@ where
                         Some("250-STARTTLS\r\n".to_string()),
                         Some("250-8BITMIME\r\n".to_string()),
                         Some("250 SMTPUTF8\r\n".to_string()),
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-server/src/receiver/pre_transaction.rs:368]: Must have `250-SMTPUTF8`

**Review by Mathieu-Lala — APPROVED** [review]

File: src/vsmtp/vsmtp-protocol/src/reader.rs (hunk 1 above)
Comment (LangArthur) [reply src/vsmtp/vsmtp-protocol/src/reader.rs:222]: removed because unused

**Review by Mathieu-Lala — APPROVED** [review]

**Review by Mathieu-Lala — DISMISSED** [review]

**Review by Mathieu-Lala — APPROVED** [review]

**Review by Mathieu-Lala — APPROVED** [review]

**Pull request merged**